src/plugins/workspace/server/saved_objects/saved_objects_wrapper_for_check_workspace_conflict.ts

```diff
@@ -0,0 +1,257 @@
+      const { overwrite, namespace } = options;
+      const bulkGetDocs = objects
+        .filter((object) => !!(object.id && options.workspaces))
+        .map((object) => {
+          const { type, id } = object;
+          /**
+           * It requires a check when overwriting objects to target workspaces
+           */
+          return {
+            type,
+            id: id as string,
+            fields: ['id', 'workspaces'],
+          };
+        });
+      const objectsConflictWithWorkspace: SavedObject[] = [];
+      const objectsMapWorkspaces: Record<string, string[] | undefined> = {};
+      if (bulkGetDocs.length) {
+        const bulkGetResult = await wrapperOptions.client.bulkGet(bulkGetDocs);
+
+        bulkGetResult.saved_objects.forEach((object) => {
+          if (!object.error && object.id && overwrite) {
+            /**
+             * When it is about to overwrite a object into options.workspace.
+             * We need to check if the options.workspaces is the subset of object.workspaces,
+             * Or it will be treated as a conflict
+             */
+            const filteredWorkspaces = SavedObjectsUtils.filterWorkspacesAccordingToSourceWorkspaces(
+              options.workspaces,
+              object.workspaces
+            );
+            const { id, type } = object;
+            if (filteredWorkspaces.length) {
+              /**
+               * options.workspaces is not a subset of object.workspaces,
+               * return a conflict error.
+               */
+              objectsConflictWithWorkspace.push({
+                id,
+                type,
+                attributes: {},
+                references: [],
+                error: {
+                  ...errorContent(SavedObjectsErrorHelpers.createConflictError(type, id)),
+                  metadata: { isNotOverwritable: true },
+                },
+              });
+            } else {
+              objectsMapWorkspaces[this.getRawId({ namespace, type, id })] = object.workspaces;
+            }
+          }
+        });
+      }
+
+      const objectsFilteredByError = objects.filter(
+        (item) =>
+          !objectsConflictWithWorkspace.find(
+            (errorItems) =>
+              this.getRawId({ namespace, type: errorItems.type, id: errorItems.id }) ===
+              this.getRawId({ namespace, type: item.type, id: item.id as string })
+          )
+      );
+      let objectsPayload = objectsFilteredByError;
+      if (overwrite) {
+        objectsPayload = objectsPayload.map((item) => {
+          if (item.id) {
+            item.workspaces =
+              objectsMapWorkspaces[
+                this.getRawId({
+                  namespace,
+                  id: item.id,
+                  type: item.type,
+                })
+              ];
+          }
+
+          return item;
+        });
+      }
+      const realBulkCreateResult = await wrapperOptions.client.bulkCreate(objectsPayload, options);
```

Comment:
Shall we check if `objectsPayload.length === 0` here? We don't need to do the bulkCreate if it's empty.

Reply:
By taking a look in the repository, bulkCreate won't do extra client call when the passed objects is empty, so I'd rather keep as it is.